diff --git a/src/bluetooth-fw/nimble/advert.c b/src/bluetooth-fw/nimble/advert.c
index 649a422..d967b69 100644
--- a/src/bluetooth-fw/nimble/advert.c
+++ b/src/bluetooth-fw/nimble/advert.c
@@ -25,6 +25,7 @@
 #include <system/passert.h>
 
 #include "nimble_type_conversions.h"
+#include <stdio.h>
 
 void bt_driver_advert_advertising_disable(void) {
   int rc;
diff --git a/src/fw/applib/wscript b/src/fw/applib/wscript
index 56854d2..8eb155b 100644
--- a/src/fw/applib/wscript
+++ b/src/fw/applib/wscript
@@ -89,6 +89,7 @@ def build(bld):
     bld.stlib(source=sources,
               target='applib',
               use=['target_includes',
+                   'libxs',
                    'libbtutil_includes',
                    'fw_includes',
                    'upng',
diff --git a/src/fw/main.c b/src/fw/main.c
index d3d77f8..fd2a716 100644
--- a/src/fw/main.c
+++ b/src/fw/main.c
@@ -171,6 +171,26 @@ static void dump_gpio_configuration_state(void) {
 }
 #endif /* DUMP_GPIO_CFG_STATE */
 
+#include "xs.h"
+extern void *modCloneMachine(void *, void*);
+extern void modRunMachineSetup(void *);
+extern void xsDeleteMachine(xsMachine *);
+
+extern void modLog_transmit(char *msg);
+
+void modLog_transmit(char *msg) {
+  PBL_LOG(LOG_LEVEL_ALWAYS, "modLog:(%s)", msg);
+}
+
+static void start_xs() {
+#if 1
+  modLog_transmit(" -- start_xs");
+  void *the = modCloneMachine(NULL, NULL);
+  modRunMachineSetup(the);
+  xsDeleteMachine(the);
+#endif
+}
+
 int main(void) {
   gpio_init_all();
 
@@ -201,6 +221,7 @@ int main(void) {
   dbgserial_init();
   pulse_early_init();
   print_splash_screen();
+  start_xs();
 
   rtc_init();
 
diff --git a/src/fw/wscript b/src/fw/wscript
index ce368a9..02a5c86 100644
--- a/src/fw/wscript
+++ b/src/fw/wscript
@@ -323,6 +323,7 @@ def _link_firmware(bld, sources):
             'root_includes',
             'startup',
             'tinymt32',
+            'libxs',
             'upng']
 
     ldscript = _generate_memory_layout(bld)
diff --git a/third_party/wscript b/third_party/wscript
index e5f5644..5ee5cea 100644
--- a/third_party/wscript
+++ b/third_party/wscript
@@ -3,6 +3,7 @@ def _recurse(ctx):
     ctx.recurse('freertos')
     ctx.recurse('nanopb')
     ctx.recurse('tinymt')
+    ctx.recurse('moddable')
 
     if ctx.env.MICRO_FAMILY.startswith('STM32'):
         ctx.recurse('hal_stm32')
